test/configuration_server_test.erl

```diff
@@ -0,0 +1,40 @@
+-module(configuration_server_test).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-include("configuration_server.hrl").
+
+default_directories_test() ->
+    ?assertEqual(?CONFIG_PATHS, configuration_server:default_directories()).
+
+default_filenames_test() ->
+    ?assertEqual(?FILE_NAMES, configuration_server:default_filenames()).
+
+choose_file_name_test() ->
+    Paths = ["/etc/tep/", "./"],
+    FileNames = ["config.ini", "tep.ini"],
+    meck:new(filelib, [unstick]),
+    meck:expect(filelib, is_regular, fun
+        (Path) when Path == "./tep.ini" -> true;
+        (_) -> false
+    end),
+    ?assertEqual(
+        {ok, "./tep.ini"},
+        configuration_server:choose_config_file(Paths, FileNames)
+    ),
+    ?assert(meck:validate(filelib)),
+    meck:unload().
```

Comment:
These tests look like they were written after the implementation, if I understand them correctly ;)
We need cases for all behaviour we want prod to exhibit. This just tests for the `./tep.ini` case, right? Since "all cases" here amounts to six instances, I think it's still reasonable to write six individual samples, but if you want we can look into looping tests (which should be easy with eunit). I don't think code generators are warranted just yet.

Reply:
This has been reworked.